### PR TITLE
Retrieve geo location database from storage system

### PR DESCRIPTION
**Pull request by lchanmann**
## Changes
- Retrieve geo location database from storage system

### CODE REVIEW

**Comment by noodlemoodle** [comment]
this looks in opt/mediatorX/journals but we don't copy build/journals files to opt when building with -u

**Comment by jlhughes** [comment]
> this looks in opt/mediatorX/journals but we don't copy build/journals files to opt when building with -u

with -u, files in build/journals are packaged into the jar file.   The db needs to be in some deployment/name path, and then included during the build under -Jname. 
Then during runtime, use: (drop the leading directory).  
x.get(Storage.class).get("GeoLite2-City.mmdb")

File: src/foam/util/geo/GeolocationSupport.js [hunk 1]
```diff
@@ -36,7 +38,14 @@ foam.CLASS({
     private final static GeolocationSupport instance__ = new GeolocationSupport();
     public static GeolocationSupport instance() {
       var ret = instance__;
-      init(foam.core.XLocator.get(), ret);
+      X x = foam.core.XLocator.get();
+      X resourceStorageX = x;
```
Comment (jlhughes) [inline src/foam/util/geo/GeolocationSupport.js]: @lchanmann can you instead modify foam/nanos/boot/Boot.java and create the appropriate Storage class there, thanks.

File: src/foam/net/ipgeo/GeolocationSupport.js [hunk 2]
```diff
@@ -86,12 +109,9 @@ foam.CLASS({
...
-          var jrlhome = System.getenv("JOURNAL_OUT");
-          File database = new File(jrlhome + "/GeoLite2-City/GeoLite2-City.mmdb");
-          try {
-            DatabaseReader dbReader = new DatabaseReader.Builder(database).build();
+          if ( support.getDbReader() != null ) {
             try {
-              CityResponse response = dbReader.city(ip);
+              CityResponse response = support.getDbReader().city(ip);
               if ( response == null ) {
                 Loggers.logger(x).error("GeolocationSupport", "Cannot find location");
```
Comment (jlhughes) [inline src/foam/net/ipgeo/GeolocationSupport.js]:
```suggestion
                Loggers.logger(x).warning("GeolocationSupport", "Cannot find location", ip);
```

**Pull request merged**